python/graphstorm/model/rgcn_encoder.py

```diff
@@ -504,3 +589,112 @@ def forward(self, g, inputs, mod_args=None, mod_kwargs=None):
                 rsts[nty] = self.agg_fn(alist, nty)
         return rsts
+
+
+class GraphConvwithEdgeFeat(nn.Module):
+    """ Graph conv module with edge feature supported in message passing computation.
+
+    .. versionadded:: 0.4.0
+        Add `GraphConvwithEdgeFeat` class in v0.4.0 to support edge feature in message
+        passing computation.
+
+    Parameters
+    ----------
+    in_feat: int
+        Input feature size.
+    out_feat: int
+        Output feature size.
+    edge_feat_mp_op: str
+        The opration method to combine source node embeddings with edge embeddings in message
+        passing. Options include `concat`, `add`, `sub`, `mul`, and `div`.
+    bias: bool
+        Whether to add bias. Default: True.
+    """
+    def __init__(
+        self,
+        in_feat,
+        out_feat,
+        edge_feat_mp_op='concat',
+        bias=True):
+        super(GraphConvwithEdgeFeat, self).__init__()
+        self.in_feat = in_feat
+        self.out_feat = out_feat
+        assert edge_feat_mp_op in BUILTIN_EDGE_FEAT_MP_OPS, 'GraphStorm only support edge' + \
+            f' message passing operation in {BUILTIN_EDGE_FEAT_MP_OPS}, bug got ' + \
+            f'{edge_feat_mp_op}.'
+        self.edge_feat_mp_op = edge_feat_mp_op
+        self.bias = bias
+
+        # weights for message passing using edge embeddings.
+        if edge_feat_mp_op in ['concat']:
+            self.weights = nn.Parameter(th.Tensor(in_feat * 2, out_feat))
+        else:
+            self.weights = nn.Parameter(th.Tensor(in_feat, out_feat))
+
+        # bias
+        if bias:
+            self.h_bias = nn.Parameter(th.Tensor(out_feat))
+            nn.init.zeros_(self.h_bias)
+
+    # pylint: disable=unused-argument
+    def forward(self, rel_graph, inputs, weight=None, edge_weight=None):
+        """ Graph conv forward computation with edge feature.
+
+        Parameters
+        ----------
+        rel_graph: DGLGraph
+            Input DGL heterogenous graph with one edge type only.
+        inputs: tuple of dict of Tensor
+            Node features for each node type in the format of {ntype: tensor}.
+        weight: dict of Tensor 
+            optional external node weight tensor. Not implemented. Reserved for future use.
+        edge_weight: Tensor 
+            optional external edge weight tensor. Not implemented. Reserved for future use.
+
+        Returns
+        -------
+        h: Tensor
+            New node embeddings for destination node type.
+        """
+        assert len(inputs) == 3, 'For using edge features in message passing, you need to ' + \
+                                 'provide 3 inputs in a tuple, the format is (src_inputs, ' + \
+                                 f'dst_inputs, edge_inputs). but got {len(inputs)} inputs.'
+        src_inputs, _, edge_inputs = inputs
+
+        rel_graph.srcdata['n_h'] = src_inputs
+        rel_graph.edata['e_h'] = edge_inputs
+        if self.edge_feat_mp_op == 'concat':
```

Comment:
In docstring we need to provide the definition of each edge_feat_mp_op.

Reply:
added.